ui/src/app/guards/sign-in-guard.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  CanActivateChild,
+  Router, RouterStateSnapshot
+} from '@angular/router';
+
+import {Observable} from 'rxjs/Observable';
+
+
+import {SignInService} from 'app/services/sign-in.service';
+
+declare const gapi: any;
+
+@Injectable()
+export class SignInGuard implements CanActivate, CanActivateChild {
+  constructor(private signInService: SignInService, private router: Router) {}
+
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
+    return new Observable((observer) => {
```

Comment:
I think this is equivalent to just `return this.signInService.user.pluck('isSignedIn');`; I'm not sure you need the wrapping / unwrapping with another Observable.  If it complains about type info, you can try `pluck<any, boolean>()`